packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -67,6 +67,26 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
     [onChange]
   );
 
+  const handleSubmit = useCallback(() => {
+    if (props.onSubmit && response) {
+      let source = props.source;
+      if (!source) {
+        const profile = medplum.getProfile();
+        if (profile) {
+          source = createReference(profile);
+        }
+      }
+      props.onSubmit({
+        ...response,
+        questionnaire: getReferenceString(questionnaire as Questionnaire),
+        subject: props.subject,
+        source,
+        authored: new Date().toISOString(),
+        status: 'completed',
+      });
+    }
+  }, [medplum, props, questionnaire, response]);
```

Comment:
This `useCallback` is going to change on every re-render since `props` will always be a different object.

You probably want to pull out those things from `props` you want at the top and add them individually to the deps array OR optionally keep the function unmemoized

```suggestion
  }, [medplum, onSubmit, source, subject, questionnaire, response]);
```

Reply:
good catch